### PR TITLE
Simplify building the decoder auxiliary columns

### PR DESCRIPTION
**Pull request by Al-Kindi-0**
## Describe your changes
This PR proposes a simplification to the way the auxiliary trace columns related to the decoder are built.

## Checklist before requesting a review
- Repo forked and branch created from `next` according to naming convention.
- Commit messages and codestyle follow [conventions](./CONTRIBUTING.md).
- Relevant issues are linked in the PR description.
- Tests added for new functionality.
- Documentation/comments updated according to changes.

### CODE REVIEW

File: processor/src/trace/decoder/mod.rs [hunk 1]
```diff
@@ -120,187 +129,593 @@ fn build_aux_col_p1<E: FieldElement<BaseField = Felt>>(
...
+    let op_code_felt = main_trace.get_op_code(i);
+    let op_code = op_code_felt.as_int() as u8;
+
+    match op_code {
+        JOIN => main_trace.get_block_hash_table_inclusion_multiplicand_join(i, alphas),
+        SPLIT => main_trace.get_block_hash_table_inclusion_multiplicand_split(i, alphas),
+        LOOP => main_trace.get_block_hash_table_inclusion_multiplicand_loop(i, alphas),
+        REPEAT => main_trace.get_block_hash_table_inclusion_multiplicand_repeat(i, alphas),
+        DYN => main_trace.get_block_hash_table_inclusion_multiplicand_dyn(i, alphas),
+        _ => E::ONE,
```
[the remark this answers is not in the document]
Comment (bobbinth) [reply processor/src/trace/decoder/mod.rs:174]: I'm actually not sure about this - but let's keep it as is for now.

File: processor/src/trace/decoder/mod.rs [hunk 2]
```diff
@@ -120,187 +129,593 @@ fn build_aux_col_p1<E: FieldElement<BaseField = Felt>>(
...
-    // keep track of the last updated row in the running product column
-    let mut result_idx = 0_usize;
+// HELPER FUNCTIONS
+// ================================================================================================
 
-    for (clk, update) in aux_trace_hints.op_group_table_hints() {
-        let clk = *clk as usize;
+struct MainTrace<'a> {
+    columns: &'a ColMatrix<Felt>,
+}
```
Comment (bobbinth) [inline processor/src/trace/decoder/mod.rs:261]: Eventually, we'd probably want to combine this with the other `MainTrace` - but I think this is fine for now.

**Review by bobbinth — COMMENTED** [review]
Looks good! Thank you! I left a couple of small comments inline.

File: processor/src/trace/decoder/mod.rs (hunk 2 above)
Comment (Al-Kindi-0) [reply processor/src/trace/decoder/mod.rs:261]: >  I'm actually not sure about this - but let's keep it as is for now.
I was not sure about the behavior with respect to `CALL` and `SYSCALL`, is this what you have in mind? And if so, should we leave a TODO/ create an issue?

Comment (Al-Kindi-0) [reply processor/src/trace/decoder/mod.rs:261]: > Eventually, we'd probably want to combine this with the other `MainTrace` - but I think this is fine for now.

Agree, once we are done with the each part, I will consolidate all of the `MainTrace` structs and methods in one place.

File: processor/src/decoder/tests.rs [hunk 3]
```diff
@@ -1149,6 +1149,7 @@ fn syscall_block() {
     // build foo procedure body
     let foo_root = CodeBlock::new_span(vec![Operation::Push(THREE), Operation::FmpUpdate]);
 
+
```
[the remark this answers is not in the document]
Comment (Al-Kindi-0) [reply processor/src/decoder/tests.rs:1152]: Done!

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! I'm going to merge this into the upstream PR.

**Pull request merged**